src/plugins/event-plugins/FetchPlugin.ts

```diff
@@ -47,14 +47,17 @@ export const FETCH_PLUGIN_ID = 'com.amazonaws.rum.fetch';
  * The fetch API is monkey patched using shimmer so all calls to fetch are intercepted. Only calls to URLs which are
  * on the allowlist and are not on the denylist are traced and recorded.
  */
-export class FetchPlugin extends MonkeyPatched implements Plugin {
-    private pluginId: string;
-    private config: HttpPluginConfig;
+export class FetchPlugin
+    extends MonkeyPatched<Window, 'fetch'>
+    implements Plugin {
+    private readonly pluginId: string;
+    private readonly config: HttpPluginConfig;
     private context: PluginContext;
 
     constructor(config?: PartialHttpPluginConfig) {
         super();
         this.pluginId = FETCH_PLUGIN_ID;
+        // TODO: fix config merge so that it is not shallow
```

Comment:
nit(blocking): Remove TODOs (here and elsewhere)

Reply:
Can do 👍🏻